### DISCUSSION TITLE
g2c for large model domains

### DISCUSSION THREAD

**Discussion by PerryShafran-NOAA** [dtcenter/METplus#1161] [File I/O] [answered]
Hi, there,

Under MET 9.1, y'all had developed for me a version of the g2c library that was to be used for a large domain; it is located currently on WCOSS under /gpfs/dell2/emc/verification/noscrub/emc.metplus/modulefiles/met/g2c_test.  Now, it seems that I am getting a Bus error in running these large domains.

I'd really like to move this to MET 10.0.0.  Should I still be using the g2c library for this?  Do you have any reason why I should be getting a bus error when running point_stat?  I hadn't seen this before until now; should be the same grid that had worked previously.

Thanks!

Perry

**Comment by jprestop** [comment, marked as answer]
Hi @PerryShafran-NOAA.  

> Do you have any reason why I should be getting a bus error when running point_stat?

What version of point_stat are you running when you get a bus error? 

> I'd really like to move this to MET 10.0.0. 

You should be able to use METv10.0 on WCOSS as described on our [METplus 4.0 Existing Builds page](https://dtcenter.org/community-code/metplus/metplus-4-0-existing-builds) without getting this error.  You should also be able to use METv10.1.0-beta2 on WCOSS as described on our [METplus 4.1 Existing Builds page](https://dtcenter.org/community-code/metplus/metplus-4-1-existing-builds) without getting this error.  These versions were built using g2c/1.6.3, which should have eliminated this problem.  

If you still experience problems with these versions please send us the command line that you are running and please send the error that you are receiving so that we can try to reproduce the problem.  Thanks!

**Reply by PerryShafran-NOAA** [reply]
I believe this worked, but please keep this open until I confirm.

**Comment by malloryprow** [comment]
@jprestop We did the NCO install with the g2c fix, correct? (/gpfs/dell1/nco/ops/nwprod/met.v9.1.3)

**Reply by PerryShafran-NOAA** [reply]
I think we did, but I guess I wasn't using that.  I've upgraded now and all seems good!

**Reply by malloryprow** [reply]
Phew! Good! I was nervous we might be having more g2c problems.

**Reply by jprestop** [reply]
@malloryprow I found an email from you from March 12, 2021 where you said, "NCO installed MET 9.1.1 in nwpara and I just tested on both Mars and Luna. It passed all tests including the new one for the g2c problems!", so hopefully, we can safely assume it was done for 9.1.3 as well.  @PerryShafran-NOAA What version were you using when you experienced the problem (perhaps we should remove that version?) and what version did you test that was successful?

**Reply by PerryShafran-NOAA** [reply]
It was indeed older (9.1, not 9.1.3), but I think that it was this or an older version, with the test g2c module, that had worked previously, so that's why I was surprised that this combination had not worked recently and was wondering why that would happen.  

It looks like it works using v10, but I need to look at it to be sure.

**Reply by PerryShafran-NOAA** [reply]
OK - I am still getting a bus error.  I am confirmed that I am using MET v 10 now, with g2c/1.6.3, and I am getting a bus error.

**Reply by jprestop** [reply]
@PerryShafran-NOAA Could you please send the following information so that I can try to reproduce the error?
- The modules you are loading
- The MET command line that is causing the error

**Reply by PerryShafran-NOAA** [reply]
It looks like I was able to get it to work.  I needed to increase the memory allocation of the job in order for this to work.  I think that was the problem - ran out of memory.

**Reply by jprestop** [reply]
@PerryShafran-NOAA Thanks! Is it ok if I mark this Discussion as answered (I unmarked it when you said you got the error with METv10.00) and lock this Discussion?

**Reply by PerryShafran-NOAA** [reply]
Yes, it looks like the job is still running and producing output.  I think you can close the discussion, but I will continue to monitor to see that things are all right.

**Reply by jprestop** [reply]
Thanks @PerryShafran-NOAA.  Please follow up here when you feel confident there are no issues so that we can lock this conversation to prevent future posts. Our team has decided that once discussions have been answered, and the answer has been confirmed, we'll lock them. We want to encourage users to ask new questions in new discussions rather than posting to old ones. Hopefully that'll make the questions/answers easier for other users to follow.

**Reply by PerryShafran-NOAA** [reply]
I think that we are good now as I no longer am having any issues.  You may lock this conversation.

**Reply by jprestop** [reply]
Thanks @PerryShafran-NOAA!  Great news!  I will go ahead and lock this conversation now.
